aws/resource_aws_mwaa_environment.go

```diff
@@ -0,0 +1,612 @@
+
+	log.Printf("[INFO] Reading MWAA Environment: %s", d.Id())
+
+	environment, err := finder.EnvironmentByName(conn, d.Id())
+
+	if err != nil {
+		if isAWSErr(err, mwaa.ErrCodeResourceNotFoundException, "") && !d.IsNewResource() {
+			log.Printf("[WARN] MWAA Environment %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("error reading MWAA Environment (%s): %w", d.Id(), err)
+	}
+
+	d.Set("airflow_configuration_options", aws.StringValueMap(environment.AirflowConfigurationOptions))
+	d.Set("airflow_version", environment.AirflowVersion)
+	d.Set("arn", environment.Arn)
+	d.Set("created_at", aws.TimeValue(environment.CreatedAt).String())
+	d.Set("dag_s3_path", environment.DagS3Path)
+	d.Set("environment_class", environment.EnvironmentClass)
+	d.Set("execution_role_arn", environment.ExecutionRoleArn)
+	d.Set("kms_key", environment.KmsKey)
+	if err := d.Set("last_updated", flattenMwaaLastUpdate(environment.LastUpdate)); err != nil {
+		return fmt.Errorf("error reading MWAA Environment (%s): %w", d.Id(), err)
+	}
+	if err := d.Set("logging_configuration", flattenMwaaLoggingConfiguration(environment.LoggingConfiguration)); err != nil {
+		return fmt.Errorf("error reading MWAA Environment (%s): %w", d.Id(), err)
+	}
+	d.Set("max_workers", environment.MaxWorkers)
+	d.Set("name", environment.Name)
+	if err := d.Set("network_configuration", flattenMwaaNetworkConfiguration(environment.NetworkConfiguration)); err != nil {
+		return fmt.Errorf("error reading MWAA Environment (%s): %w", d.Id(), err)
+	}
+	d.Set("plugins_s3_object_version", environment.PluginsS3ObjectVersion)
+	d.Set("plugins_s3_path", environment.PluginsS3Path)
+	d.Set("requirements_s3_object_version", environment.RequirementsS3ObjectVersion)
+	d.Set("requirements_s3_path", environment.RequirementsS3Path)
+	d.Set("service_role_arn", environment.ServiceRoleArn)
+	d.Set("source_bucket_arn", environment.SourceBucketArn)
+	d.Set("status", environment.Status)
+	d.Set("webserver_access_mode", environment.WebserverAccessMode)
+	d.Set("webserver_url", environment.WebserverUrl)
+	d.Set("weekly_maintenance_window_start", environment.WeeklyMaintenanceWindowStart)
+
+	if err := d.Set("tags", keyvaluetags.MwaaKeyValueTags(environment.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsMwaaEnvironmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mwaaconn
+
+	input := mwaa.UpdateEnvironmentInput{
+		Name: aws.String(d.Get("name").(string)),
+	}
+
+	if d.HasChangesExcept("tags") {
+		options, ok := d.GetOk("airflow_configuration_options")
+		if !ok {
+			options = map[string]interface{}{}
+		}
+
+		input.AirflowConfigurationOptions = stringMapToPointers(options.(map[string]interface{}))
+
+		input.AirflowVersion = aws.String(d.Get("airflow_version").(string))
```

Comment:
Although it doesn't seem strictly necessary in this case, it is good practice to only send an attribute in the update if it has changed. i.e. add back `d.HasChange()` around each potentially changed attribute:

```go
if d.HasChange("airflow_version") {
    input.AirflowVersion = aws.String(d.Get("airflow_version").(string))
}
```

Reply:
@ewbankkit I fixed it, so would you mind reviewing the pr again?